source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionOffer/add/AddDispositionOfferContainer.tsx

```diff
@@ -1,11 +1,76 @@
+import { AxiosError } from 'axios';
+import { useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
+import { useDispositionProvider } from '@/hooks/repositories/useDispositionProvider';
+import { useModalContext } from '@/hooks/useModalContext';
+import { IApiError } from '@/interfaces/IApiError';
+import { Api_DispositionFileOffer } from '@/models/api/DispositionFile';
+
+import { IDispositionOfferFormProps } from '../form/DispositionOfferForm';
+import { DispositionOfferFormModel } from '../models/DispositionOfferFormModel';
+
 export interface IAddDispositionOfferContainerProps {
   dispositionFileId: number;
+  View: React.FC<IDispositionOfferFormProps>;
 }
 
 const AddDispositionOfferContainer: React.FunctionComponent<
   React.PropsWithChildren<IAddDispositionOfferContainerProps>
-> = ({ dispositionFileId }) => {
-  return <p>Add Disposition Offer Works!</p>;
+> = ({ dispositionFileId, View }) => {
+  const history = useHistory();
+  const location = useLocation();
+
+  const { setModalContent, setDisplayModal } = useModalContext();
+  const [offerStatusError, setOfferStatusError] = useState(false);
+
+  const backUrl = location.pathname.split('/offers/new')[0];
```

Comment:
I'm not sure I love this pattern. Is this what we are doing in acquisition?

Reply:
It was used for Form8